### PR TITLE
Add Text-Input

### PR DESCRIPTION
**Pull request by Dhruv-Rathi**

### CODE REVIEW

File: src/Base/index.js [hunk 1]
```diff
@@ -1,78 +1,77 @@
...
-            ? props.onPressColor || '#27272A'
-            : props.backgroundColor || '#18181B',
-          width: '100%',
-          borderColor: props.borderColor || '#27272A',
+    <View style={{paddingVertical: 12}}>
+      {label && <Text>{label}</Text>}
+
+      <View
+        style={{
+          height: 42,
```
Comment (rjvim) [inline src/Base/index.js]: @Dhruv-Rathi User should be able to override these styles, passing them as props, that's why in button component i used something like:

```
            ...(props.loaderIconWrapperStyles || {
              alignItems: 'center',
              flexDirection: 'row',
              justifyContent: 'center',
            }),
```

**Review by rjvim — CHANGES_REQUESTED** [review]

**Comment by Dhruv-Rathi** [comment]
@rjvim  please review.

File: src/Base/index.js [hunk 2]
```diff
@@ -1,78 +1,93 @@
...
-            props.children
-          )}
-        </View>
-      )}
+    <View style={{paddingVertical: 12}}>
+      {label && <Text>{label}</Text>}
+
+      <View
+        style={{
+          ...(props.textInputStyles || {
```
Comment (rjvim) [inline src/Base/index.js]: I think would be called - wrapperStyles, as this View is wrapping TextInput

File: src/Base/index.js [hunk 3]
```diff
@@ -1,78 +1,93 @@
...
+  style,
+  value,
+  placeholder = 'Enter Text',
+  label = null,
+  error,
   ...props
 }) => {
+  const [focused, setFocused] = React.useState(false);
   const textStyles = {
     ...(props.textStyles || {
```
Comment (rjvim) [inline src/Base/index.js]: This should be renamed to props.textInputStyles

The textStyles name is used in buttons package, is literally for <Text>, here we have TextInput, so naming of this is going to be important

**Review by rjvim — CHANGES_REQUESTED** [review]

**Comment by Dhruv-Rathi** [comment]
@rjvim please check now.

File: src/Base/index.js [hunk 4]
```diff
@@ -1,78 +1,93 @@
...
-      {busy === false && (
-        <View>
-          {typeof props.children === 'string' ? (
-            <Text style={textStyles}>{props.children}</Text>
-          ) : (
-            props.children
-          )}
-        </View>
-      )}
+    <View style={{paddingVertical: 12}}>
```
Comment (rjvim) [inline src/Base/index.js]: Only one thing left - `paddingVertical: 12` 

This would become hardcoded

Comment (rjvim) [reply src/Base/index.js]: @Dhruv-Rathi

Comment (Dhruv-Rathi) [reply src/Base/index.js]: `<View style={{...(props.labelStyles || {paddingVertical: 12})}}>`
@rjvim  This is fine?

Comment (rjvim) [reply src/Base/index.js]: @Dhruv-Rathi I think there are not label styles but the styles for entire Wrapper. I think we should put any styles at all for this. 

What I mean is, first of all, padding or margin what's needed would be the issue of the screen which will use our `<TextInput />` component. So, someone who wants to use our component, should wrap it in something like:

```
<View style={{ marginVertical: 12 }}>
<TextInput />
</View>
```


Is it possible to entirely remove "View" wrapping?

**Comment by Dhruv-Rathi** [comment]
@rjvim is it fine now?

**Comment by Dhruv-Rathi** [comment]
@rjvim Please review.

**Pull request merged**